### PR TITLE
Feature/jo 2/dev humna/populate reddit data into html table

### PR DESCRIPTION
**Pull request by HumnaAleem1**

### CODE REVIEW

File: src/components/Home/Home.tsx [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
-        console.log(response)
+        const response: AxiosResponse = await get(redditUrl)
+        setData(response?.data?.data?.children)
     }
 
     return (
     <div>
         <div>Test App</div>
         <button onClick={getRedditData}>Go</button>
+        { data && <Table data={data}/> }
```
Comment (junaidrt) [inline src/components/Home/Home.tsx]: Please use appropriate prop name instead of data.

File: src/Routes.js [hunk 2]
```diff
@@ -1,13 +1,11 @@
 import { Switch, Route } from 'react-router-dom';
 import { Home } from './components/Home/Home'
 
-const Routes = () => {
+export const Routes = () => {
```
Comment (hamzamubeen-sl2) [inline src/Routes.js]: since it is a component why not make it default export ?

File: src/components/Home/Home.tsx [hunk 3]
```diff
@@ -1,17 +1,22 @@
+import { useState } from 'react'
 import { redditUrl } from '../../API/Constants'
 import { get } from '../../API/API'
+import { Table } from '../Table/Table'
```
Comment (hamzamubeen-sl2) [inline src/components/Home/Home.tsx]: why is it `Table/Table`

**Review by hamzamubeen-sl2 — CHANGES_REQUESTED** [review]

File: src/components/Home/Home.tsx (hunk 3 above)
Comment (HumnaAleem1) [reply src/components/Home/Home.tsx]: Junaid said .... avoid to use index named files. component should be a proper name.

Comment (hamzamubeen-sl2) [reply src/components/Home/Home.tsx]: ok then we follow snake-case naming convention for directories

Comment (HumnaAleem1) [reply src/components/Home/Home.tsx]: oh... Ok

File: src/RedditInterfaces.ts [hunk 4]
```diff
@@ -1,20 +1,20 @@
...
 
-interface Children {
+interface IChildren {
     kind: string,
-    data: ChildData
+    data: IChildData
 }
 
-interface ChildData {
+interface IChildData {
```
Comment (hamzamubeen-sl2) [inline src/RedditInterfaces.ts:17]: i don't understand what are these types for totally ambiguous.

File: src/components/Card/Card.tsx [hunk 5]
```diff
@@ -1,12 +1,10 @@
...
 
-interface Props {
-	authorName: string
-    title: string
-}
-
-const Card: React.FC<Props> = ({authorName, title}) => {
+interface ICardProps {}
 
+export const Card: React.FC<ICardProps> = ({ ...tableStates }) => {
```
Comment (hamzamubeen-sl2) [inline src/components/Card/Card.tsx]: you dont need to Write React.FC   you can simply import and use FC.

File: src/components/Table/Table.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+import { IRedditData } from "../../RedditInterfaces"
+import './Table.css'
+
+interface ITableProps {
+    authorData: IRedditData['data']['children']
+}
+
+export const Table: React.FC<ITableProps> = ({ authorData }) => {
+
+    const [state, setState] = useState<ITableStates>({
```
Comment (hamzamubeen-sl2) [inline src/components/Table/Table.tsx]: i think it is not appropriate to use names such as state and setState

**Review by hamzamubeen-sl2 — CHANGES_REQUESTED** [review]

File: src/RedditInterfaces.ts (hunk 4 above)
Comment (HumnaAleem1) [reply src/RedditInterfaces.ts:17]: you can ignore it. It was just a practice.

File: src/components/Table/Table.tsx (hunk 6 above)
Comment (HumnaAleem1) [reply src/components/Table/Table.tsx]: Ok I can rename.

Comment (hamzamubeen-sl2) [reply src/components/Table/Table.tsx]: also it is not recommended to use a single useState to keep all the state data in a FC.
read this: [https://stackoverflow.com/questions/58607755/react-hooks-usestate-what-recommended-have-just-one-state-per-component-or-mu](url)

File: src/RedditInterfaces.ts (hunk 4 above)
Comment (junaidrt) [reply src/RedditInterfaces.ts:17]: @HumnaAleem1 Can we remove all the redundant types defined here ?

**Review by junaidrt — CHANGES_REQUESTED** [review]

**Review by junaidrt — APPROVED** [review]

**Review by hamzamubeen-sl2 — APPROVED** [review]

**Pull request merged**